module/modmanager/manager.go

```diff
@@ -70,7 +70,7 @@ type module struct {
 	dataDir   string
 	process   pexec.ManagedProcess
 	handles   modlib.HandlerMap
-	conn      *grpc.ClientConn
+	conn      rdkgrpc.ReconfigurableClientConn
```

Comment:
Why not a pointer?

Reply:
copied how robot client does it, but conceptually I also wanted to tie the connection to the lifetime of the module struct, so making it not a pointer made sense

Reply:
Gotcha; yeah I wish Go gave better interiority on what the "lifetime" of something is. In the end, I'm not sure it matters too much whether this field is a pointer or a value. 

But, even if it's a value, that value contains an interface `rpc.ClientConn` which is inherently a pointer. So, I think passing around the `rdkgrpc.ReconfigurableClientConn` value (copying) could still keep references to that `rpc.ClientConn` around, so I'm not sure keeping `conn` as a value here ensures much about the lifetime of the underlying connection.

Reply:
Per that recent "garbage collection" talk in the reading group, though, Go's GC system is not really based on reference counting, so I don't really know what's going on under the hood here 🤷🏻 . Anyway, feel free to keep as a value.

Reply:
yep, I don't think it matters too much ultimately, but would like to use references when possible